src/VisualStudio/Core/Def/Implementation/VirtualMemoryNotificationListener.cs

```diff
@@ -1,183 +0,0 @@
-    private bool _infoBarShown;
-
-    private VirtualMemoryNotificationListener(
-        IVsShell shell,
-        IGlobalOptionService globalOptions,
-        VisualStudioWorkspace workspace)
-    {
-        _globalOptions = globalOptions;
-        _workspace = workspace;
-        _workspaceCacheService = workspace.Services.GetService<IWorkspaceCacheService>() as WorkspaceCacheService;
-
-        if (GCSettings.IsServerGC)
-        {
-            // Server GC has been explicitly enabled, which tends to run with higher memory pressure than the
-            // default workstation GC. Allow this case without triggering frequent feature shutdown.
-            return;
-        }
-
-        _workspace.WorkspaceChanged += OnWorkspaceChanged;
-
-        // Note: We never unhook this event sink. It lives for the lifetime of the host.
-        ErrorHandler.ThrowOnFailure(shell.AdviseBroadcastMessages(this, out var cookie));
-    }
-
-    public static async Task<VirtualMemoryNotificationListener> CreateAsync(
-        VisualStudioWorkspace workspace,
-        IThreadingContext threadingContext,
-        IAsyncServiceProvider serviceProvider,
-        IGlobalOptionService globalOptions,
-        CancellationToken cancellationToken)
-    {
-        await threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
-
-        var shell = (IVsShell?)await serviceProvider.GetServiceAsync(typeof(SVsShell)).ConfigureAwait(true);
-        Assumes.Present(shell);
-
-        return new VirtualMemoryNotificationListener(shell, globalOptions, workspace);
-    }
-
-    /// <summary>
-    /// Called by the Visual Studio Shell to notify components of a broadcast message.
-    /// </summary>
-    /// <param name="msg">The message identifier.</param>
-    /// <param name="wParam">First parameter associated with the message.</param>
-    /// <param name="lParam">Second parameter associated with the message.</param>
-    /// <returns>S_OK always.</returns>
-    public int OnBroadcastMessage(uint msg, IntPtr wParam, IntPtr lParam)
-    {
-        switch (msg)
-        {
-            case VSConstants.VSM_VIRTUALMEMORYLOW:
-            case VSConstants.VSM_VIRTUALMEMORYCRITICAL:
-            case VSConstants.VSM_MEMORYHIGH:
-            case VSConstants.VSM_MEMORYEXCESSIVE:
-                {
-                    if (!_alreadyLogged)
-                    {
-                        // record that we had hit critical memory barrier
-                        Logger.Log(FunctionId.VirtualMemory_MemoryLow, KeyValueLogMessage.Create(m =>
-                        {
-                            // which message we are logging and memory left in bytes when this is called.
-                            m["MSG"] = msg;
-                            m["MemoryLeft"] = (long)wParam;
-                        }));
-
-                        _alreadyLogged = true;
-                    }
-
-                    _workspaceCacheService?.FlushCaches();
```

Comment:
This might be removing any remaining value of the IWorkspaceCacheService generally. The only remaining use I see is in AbstractImportCompletionCacheServiceFactory which subscribes to the cache flush. That might be a no-op now too.

Reply:
i remove it in a nother PR :)